src/lib/components/button.svelte

```diff
@@ -12,15 +12,16 @@
     on:click
     class:selected={active}
     {disabled}
-    class={`${size} ${variant}`}><slot /></button
+    class={`flex items-center justify-center ${size} ${variant}`}
```

Comment:
Do these work with Tailwind's tree-shaking/purging?

Reply:
Also: Svelte allows you to interpolate with double-quoted strings. (e.g. `class="flex items-center justify-center {size} {variant}"`, but I'm still hesitant about the tree shacking. I'd probably use Svelte's [class toggles](https://svelte.dev/repl/e3ea17e8e09044999bf7cb4bc882adea?version=3.19.2).

Reply:
how would it work since its different styles like for size it could be small or large?

Reply:
So, this is a bit of an exercise in API design.

Above, you have:

```ts
export let size: string = 'small';
```

This is a little problematic because the consumer can pass in _any_ string.

You _could_ tighten the types up a bit.

```ts
export let size: 'large' | 'small' = 'small';
```

No were down to only two options. You'd do the same with `variant` and `as` as well.

You could now do something like this:

```svelte
<button
  on:click
  class:selected={active}
  {disabled}
  class="flex items-center justify-center"
  class:small={variant === 'small'}
  class:large={variant === 'large'}
>
  <slot />
</button>
```

The other option, of course since "small" is the default, just allowing a `large` boolean prop that overrides the default.

```html
<Button large>Hello</Button>
```

And now, you can do something like this:

```svelte
<button
  on:click
  class:selected={active}
  {disabled}
  class="flex items-center justify-center"
  class:large
>
  <slot />
</button>
```

The pattern only works for two settings where one is the default. So, if you suspect there will be more, than you're going to need to stick with the `variant` pattern.